src/mlpack/tests/q_learning_test.cpp

```diff
@@ -170,4 +171,74 @@ BOOST_AUTO_TEST_CASE(CartPoleWithDoubleDQN)
   BOOST_REQUIRE(converged);
 }
 
+//! Test DQN in Acrobat task.
+BOOST_AUTO_TEST_CASE(AcrobatWithDQN)
+{
+  // Set up the network.
+  FFN<MeanSquaredError<>, GaussianInitialization> model(MeanSquaredError<>(),
+      GaussianInitialization(0, 0.001));
+  model.Add<Linear<>>(4, 64);
+  model.Add<ReLULayer<>>();
+  model.Add<Linear<>>(64, 32);
+  model.Add<ReLULayer<>>();
+  model.Add<Linear<>>(32, 3);
+
+  // Set up the policy and replay method.
+  GreedyPolicy<Acrobat> policy(1.0, 1000, 0.1);
+  RandomReplay<Acrobat> replayMethod(20, 10000);
+
+  TrainingConfig config;
+  config.StepSize() = 0.01;
+  config.Discount() = 0.99;
+  config.TargetNetworkSyncInterval() = 100;
+  config.ExplorationSteps() = 100;
+  config.DoubleQLearning() = false;
+  config.StepLimit() = 400;
+
+  // Set up DQN agent.
+  QLearning<Acrobat, decltype(model), AdamUpdate, decltype(policy)>
+      agent(std::move(config), std::move(model), std::move(policy),
+          std::move(replayMethod));
+
+  arma::running_stat<double> averageReturn;
+  size_t episodes = 0;
+  bool converged = true;
+  while (true)
+  {
+    double episodeReturn = agent.Episode();
+    averageReturn(episodeReturn);
+    episodes += 1;
+
+    if (episodes > 1000)
+    {
+      Log::Debug << "Acrobat with DQN failed." << std::endl;
+      converged = false;
+      break;
+    }
+
+    /**
+     *
+     * I am using a thresold of -350.0 . While testing it my previous code 
+     * on Q learning https://github.com/luffy1996/rl-using-mlpack/blob/master/qlearning.cpp
+     * I witnessed that -350.0 is average return after 80 episodes.
+     * https://drive.google.com/open?id=1QzZP5egUWhcSJr_byceihU8rIF5sSqdU
+     * Hence I am using -350.0 to test convergence.
```

Comment:
The 'goal' is to swing the end to a height of at least one link above the base, over a couple of runs, correct? I'm not sure -350.0 does represent that state. 

So not sure where 50 is coming from:

``size_t requirement = 50; // This variable checks if the game is converging or not .``